workflow/controller/controller.go

```diff
@@ -1053,42 +1017,62 @@ func (wfc *WorkflowController) newPodInformer(ctx context.Context) cache.SharedI
 }
 
 func (wfc *WorkflowController) newConfigMapInformer() cache.SharedIndexInformer {
-	indexInformer := v1.NewFilteredConfigMapInformer(wfc.kubeclientset, wfc.GetManagedNamespace(), 20*time.Minute, cache.Indexers{
-		indexes.ConfigMapLabelsIndex: indexes.ConfigMapIndexFunc,
-	}, func(opts *metav1.ListOptions) {
-		opts.LabelSelector = common.LabelKeyConfigMapType
+
+	indexInformer := v1.NewConfigMapInformer(wfc.kubeclientset, wfc.GetManagedNamespace(), 20*time.Minute, cache.Indexers{
```

Comment:
So I mentioned [before](https://github.com/argoproj/argo-workflows/pull/11855#discussion_r1342787539) that there are _two_ namespaces to watch, and both are used for different purposes. This is on the managed namespace, which is not necessarily the same as Argo's own namespace. The `workflow-controller-configmap` will _always_ be in Argo's own namespace, but semaphores etc may be in a separate namespace. As in, if you deploy Argo in one namespace but run Workflows in a _different_ namespace(s).

So we may very well need two separate Informers to handle both namespace configurations, unless you can listen to both namespaces in one.

Reply:
As far as I know, we can only watch 1 or all namespaces, so I think we have 2 possibilities (based on my limited knowledge):
#1. Create another function and duplicate the code;
#2. Create an array with all necessary namespaces and move the current code into this loop;

So far I would prefer #2, but I'm not sure if it may cause some problem